h/util/group_scope.py

```diff
@@ -28,6 +30,30 @@ def uri_scope(uri):
     return _parse_origin(uri)
 
 
+def uri_to_scope(uri):
+    """
+    Return a tuple representing the origin and path of a URI
+
+    :arg uri: The URI from which to derive scope
+    :type uri: str
+    :rtype: tuple(str, str or None)
+    """
+    # A URL with no origin component will result in a `None` value for
+    # origin, while a URL with no path component will result in an empty
+    # string for path.
+    origin = _parse_origin(uri)
+    path = _parse_path(uri) or None
+    return (origin, path)
+
+
+def _parse_path(uri):
+    """Return the path component of a URI string"""
+    if uri is None:
+        return None
```

Comment:
Is `uri` ever `None` here? The tests still pass if I replace the function with `return urlparse.urlsplit(uri)[2]`, and I can't see from the code where `None` might be passed.